js/src/forum/states/DiscussionListState.js

```diff
@@ -88,6 +88,11 @@ export default class DiscussionListState {
 
     return this.loadResults().then(
       (results) => {
+        // If `this.loading` is false, that means that a prior request loaded while this one was in process.
+        // As such, before we add on the discussion items we just loaded in, we want to clear the list again.
+        if (clear && !this.loading) {
```

Comment:
I'd keep the `clear: false` option and just remove the check on this line to always clear the discussions. This way calling `refresh({clear: false})` could refresh the `DiscussionList` without the loading animation. Could be quite useful for when you unfollow a discussion, the `DiscussionList` could be updated in a smooth way by calling this method.

Reply:
I like that approach! Done